src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -291,7 +291,7 @@ export function isKernelRegisteredByUs(kernelSpec: IJupyterKernelSpec): 'oldVers
         return 'newVersion';
     }
     const guidRegEx = /[a-f0-9]{32}$/;
-    if (kernelSpec.metadata?.interpreter && kernelSpec.name.toLowerCase().search(guidRegEx)) {
+    if (kernelSpec.metadata?.interpreter && kernelSpec.name.toLowerCase().search(guidRegEx) !== -1) {
```

Comment:
This bug that I fixed was messing up the tests, but I believe that it wasn't causing actual product code issues as I think that only our extension puts the interpreter into the metadata. So even though the guid check was always returning truthy (-1 != false) it wasn't flagging anything as registered by us that wasn't registered by us.